### PR TITLE
Convert to Sphinx theme & Python package

### PR DESCRIPTION
**Pull request by ajschmidt8**
This PR converts the existing implementation into a Sphinx theme & Python package.

The theme itself lives in the `rapidsai_sphinx_theme` directory.

The `rapidsai_sphinx_theme/__init__.py` file is where global RAPIDS configuration options can be set.

The `rapidsai_sphinx_theme/static` directory contains custom CSS and JS that are injected into the generated HTML files.

The `docs` directory contains some example documentation that can be built using the theme.

### CODE REVIEW

**Comment by ajschmidt8** [comment]
Some additional thoughts:

- The version switcher JSON files can be generated dynamically and published to URLs like `docs.rapids.ai/_versions/cudf.json`. We'll later switch this to just `rapids.ai/_versions/cudf.json` when we merge the sites.
- Generating and publishing these files is trivial and can be finished in less than a day.
- Since the version dropdown JSON file requires absolute, fully qualified URLs (as mentioned in the [docs](https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/version-dropdown.html#configure-switcher-json-url)), I don't think we need to add any of the crazy conditional logic that we've spoken about in the past. We can use it as-is.

**Comment by ajschmidt8** [comment]
~~Another thought... we can also hook up Netlify to this repository pretty easily so that the demo content in the `docs` directory gets published to preview environments on each PR.~~

Well...nevermind. Netlify's Python version is too old to support Sphinx builds :expressionless: https://answers.netlify.com/t/the-current-python-version-3-8-is-very-old/96966

File: README.md [hunk 1]
```diff
@@ -1,23 +1,41 @@
...
+pip install git+https://github.com/rapidsai/sphinx-theme.git@main
+```
+
+Setup the `conf.py` file:
+
+```py
+html_theme = "rapidsai_sphinx_theme"
+```
 
-## Navigation Standardization
```
Comment (exactlyallan) [inline README.md:16]: I'll move this type of content into another readme

File: docs/index.rst
Comment (exactlyallan) [inline docs/index.rst:1]: What will this section be used for?

**Review by exactlyallan — APPROVED** [review]

File: docs/index.rst
Comment (ajschmidt8) [reply docs/index.rst:1]: This section will just be used to demo doc changes.

So for instance if we add a new layout or override a layout from the PyData theme, we can build the docs in the `docs/` folder to see how it looks.

It looks like PyData uses this heavier duty tool for local theme development: https://github.com/pradyunsg/sphinx-theme-builder

but that might be overkill for our initial implementation. We can always add that later if we want.

File: README.md (hunk 1 above)
Comment (ajschmidt8) [reply README.md:16]: Sounds good. These links are hardcoded into the theme in `rapidsai_sphinx_theme/__init__.py`, therefore I didn't think this content was necessary.

**Pull request merged**